src/store/sagas/wallets.ts

```diff
@@ -652,6 +652,26 @@ export function* addSigningDeviceWorker({
     );
     let signersToUpdate = [];
 
+    try {
+      // update signers with signer count
+      let incrementForCurrentSigners = 0;
+      signers = signers.map((signer) => {
+        if (signer.type === SignerType.MY_KEEPER) {
+          const myAppKeys = filteredSigners.filter((s) => s.type === SignerType.MY_KEEPER);
+          const currentInstanceNumber = WalletUtilities.getInstanceNumberForSigners(myAppKeys);
+          const instanceNumberToSet = currentInstanceNumber + incrementForCurrentSigners;
+          signer.extraData = { instanceNumber: instanceNumberToSet + 1 };
```

Comment:
Now I think here: https://github.com/bithyve/bitcoin-keeper/blob/63306c5868beba34f8242a622df93b53edb9109f/src/store/sagas/upgrade.ts#L399 we can skip instanceNumber update for keys where instanceNumber already exists.

Reply:
I was referring to change the code in [bitcoin-keeper/src/store/sagas/upgrade.ts](https://github.com/bithyve/bitcoin-keeper/blob/63306c5868beba34f8242a622df93b53edb9109f/src/store/sagas/upgrade.ts#L399) itself so it won't change instanceNumber if it already exists